File: client/src/types/widget.ts
export type WidgetType = 'Text' | 'LineChart' | 'BarChart';

export type ChartPoint = {
  order: number;
  label: string;
  value: number;
};

export type Widget = {
  id: string;
  type: WidgetType;
  position: number;
  text?: string | null;
  points?: ChartPoint[] | null;
  createdAt: string;
  updatedAt?: string | null;
};

export type WidgetStatus = 'idle' | 'loading' | 'error';

export const widgetTypeLabel: Record<WidgetType, string> = {
  Text: 'Text',
  LineChart: 'Line chart',
  BarChart: 'Bar chart',
};

File: backend/YouScanDashboard.Api/Database/Entities/Widget.cs
namespace YouScanDashboard.Api.Database.Entities;

public enum WidgetType { Text, LineChart, BarChart }

public class ChartPoint
{
    public required int Order { get; set; }
    public required string Label { get; set; }
    public required int Value { get; set; }
}

public class Widget
{
    public Guid Id { get; set; }
    public required WidgetType Type { get; init; }
    public required int Position { get; set; }

    public string? Text { get; set; }
    public List<ChartPoint>? Points { get; set; }
    
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}